### ISSUE TITLE
[BUG] Azure.Security.KeyVault.Secrets preview2 doesn't support update

### ISSUE THREAD

**Issue by KSchlobohm** [Azure/azure-sdk-for-net#7189]
**Describe the bug**
Following along with the [documentation](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md) the process for updating a secret is [shown](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md#update-an-existing-secret). This process does not compile (the Secret constructor requires 2 string params) and does not succeed due to a required parameter of secret.Version which is a read-only property.

NOTE: another issue with the example is that the returned object from client.Update is of type SecretBase which does not have a "Value" property as shown in the example. This "missing property" issue also applies to the [delete example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md#delete-a-secret) in the Readme.md file.

***Exception or Stack Trace***
- System.ArgumentNullException: 'Value cannot be null. (Parameter 'secret.Version')'

**To Reproduce**

1. [Example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md#update-an-existing-secret)

***Code Snippet***
```cs
Secret secret = new Secret("secret-name");

// Clients may specify the content type of a secret to assist in interpreting the secret data when it's retrieved
secret.ContentType = "text/plain";

// You can specify additional application-specific metadata in the form of tags.
secret.Tags["foo"] = "updated tag";

SecretBase updatedSecret = client.Update(secret);

Console.WriteLine(updatedSecret.Name);
Console.WriteLine(updatedSecret.Value);
Console.WriteLine(updatedSecret.Version);
Console.WriteLine(updatedSecret.ContentType);
```

**Expected behavior**
Expecting to be able to set the version of the secret and receive an updatedSecret response.

**Screenshots**

**Setup (please complete the following information):**
 - OS: [Windows 10]
 - IDE : [Visual Studio]
 - Version of the Library used: Azure.Security.KeyVault.Secrets -Version 4.0.0-preview.2

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by KSchlobohm** [comment]
Looks like the link to the KeyVault Readme is [changing](https://github.com/Azure/azure-sdk-for-net/pull/7205) with regard to the link for Steps to Reproduce.

**Comment by AlexGhiondea** [comment]
@KSchlobohm thanks for pointing this out! We'll take a look at this!

**Comment by heaths** [comment]
The client code was fixed in #7729. Will leave open to track making sure the samples are updated.
